dpks/plot.py

```diff
@@ -1,7 +1,133 @@
 import seaborn as sns  # type: ignore # noqa: F401
+from typing import TYPE_CHECKING, Any, Union
+
+import numpy as np
+import pandas as pd
+import matplotlib
+import sklearn
+
+import matplotlib.pyplot as plt
+from matplotlib import cm
+
+
+if TYPE_CHECKING:
+    from .quant_matrix import QuantMatrix
+else:
+    QuantMatrix = Any
 
 
 class Plot:
+    """the base class"""
+
     def __init__(self) -> None:
         """init"""
+
+        pass
+
+    def plot(self) -> None:
+        """create the plot"""
         pass
+
+
+class SHAPPlot(Plot):
+    def __init__(
+        self,
+        fig,
+        ax,
+        shap_values: np.ndarray,
+        X: np.ndarray,
+        qm: QuantMatrix,
+        cmap: Union[list, str],
+        n_display: int = 5,
+    ) -> plt.Figure:
+        """Creates a SHAP summary plot-like figure.
+
+        Args:
+            shap_values (np.ndarray): shap values
+            X (np.ndarray): feature values
+            qm (QuantMatrix): quantmatrix
+
+        Returns:
+            plt.Figure: figure object
+        """
+        assert shap_values.shape[0] == X.shape[0]
```

Comment:
Replace with a conditional and a meaningful error message if we want to keep this in, otherwise remove